### PR TITLE
[IMPROVE] Allow override of default department for SMS Livechat sessions

### PR DESCRIPTION
**Pull request by bhardwajaditya**
## Proposed changes (including videos or screenshots)
-  Allow override of default department for SMS Livechat sessions
- Current implementation searches for room for SMS sessions using just visitor tokens, updated it to use department ID as well so that a user can have different rooms open with different departments.
- Updated admin settings description for SMS endpoint

image

## Issue(s)

https://github.com/RocketChat/feature-requests/issues/527

## Steps to test or reproduce

Modify the post endpoint URL in twilio WhatsApp configuration as **https://<SERVER_URL>/api/v1/livechat/sms-incoming/twilio?departmentName=<departmentName>** to test.

### CODE REVIEW

File: app/livechat/imports/server/rest/sms.js [hunk 1]
```diff
@@ -45,9 +45,8 @@ const defineVisitor = (smsNumber) => {
 		});
 	}
```
Comment (KevLehman) [inline app/livechat/imports/server/rest/sms.js:47]: Can you please validate if the department exists before assigning it to the visitor? 👀

**Review by KevLehman — CHANGES_REQUESTED** [review]

**Comment by KevLehman** [comment]
Hey, thanks for this contrib! 🥳 i requested a small change, let me know if you are available to do it or if I should do it, thanks!

**Comment by bhardwajaditya** [comment]
@KevLehman thanks for the review. Fixed it 😊

File: app/livechat/imports/server/rest/sms.js (hunk 1 above)
Comment (bhardwajaditya) [reply app/livechat/imports/server/rest/sms.js:47]: Fixed this at line 74, that will ensure wrong department never reaches here

**Review by KevLehman — APPROVED** [review]

**Pull request merged**